nncf/openvino/quantization/cache_statistics.py

```diff
@@ -0,0 +1,136 @@
+# Copyright (c) 2024 Intel Corporation
```

Comment:
Could you explain reasons why you decided not implement cache statistics as a cross backend method?

Reply:
I moved the caching logic into a shared function instead of making it a class with backend methods. I also added a preprocessing call for OpenVINO backend. I think the difference between backend should be only in model preprocessing step

What do you think?